src/foremast/elb/format_listeners.py

```diff
@@ -118,26 +118,63 @@ def format_listeners(elb_settings=None, env='dev'):
     return listeners
 
 
-def format_cert_name(account='', certificate=None):
+def format_cert_name(env='', account='', certificate=None):
     """Format the SSL certificate name into ARN for ELB.
 
     Args:
-        account (str): Account number for ARN.
-        certificate (str): Name of SSL certificate.
+        env (str): Account environment name
+        account (str): Account number for ARN
+        certificate (str): Name of SSL certificate
 
     Returns:
-        None: Certificate is not desired.
-        str: Fully qualified ARN for SSL certificate.
+        str: Fully qualified ARN for SSL certificate
+        None: Certificate is not desired
     """
     cert_name = None
 
     if certificate:
         if certificate.startswith('arn'):
             cert_name = certificate
         else:
-            cert_name = ('arn:aws:iam::{account}:server-certificate/'
-                         '{name}'.format(account=account,
-                                         name=certificate))
+            generated_cert_name = generate_custom_cert_name(env, account, certificate)
+            if generated_cert_name:
+                cert_name = generated_cert_name
+            else:
+                cert_name = ('arn:aws:iam::{account}:server-certificate/{name}'.format(account=account,
+                                                                                       name=certificate))
     LOG.debug('Certificate name: %s', cert_name)
 
     return cert_name
+
+
+def generate_custom_cert_name(env='', account='', certificate=None):
+    """Generate a custom TLS Cert name based on a template.
+
+    Args:
+        env (str): Account environment name
+        account (str): Account number for ARN.
+        certificate (str): Name of SSL certificate.
+
+    Returns:
+        str: Fully qualified ARN for SSL certificate.
+        None: Template doesn't exist.
+    """
+    cert_name = None
+    template_kwargs = {
+        'account': account,
+        'name': certificate
+    }
+
+    # TODO: Investigate moving this to a remote API, then fallback to local file if unable to connect
+    try:
+        rendered_template = get_template(template_file='infrastructure/tlscert_naming.json.j2', **template_kwargs)
+    except ForemastError:
+        LOG.info('Unable to find TLS Cert Template; Falling back to default logic...')
+        return cert_name
+
+    try:
+        cert_name = json.loads(rendered_template)[env][certificate]
+    except KeyError:
+        LOG.error("Unable to find TLS certificate named {0} under {1} in TLS Cert Template".format(certificate, env))
```

Comment:
Use lazy evaluation for logger to match convention with rest of Foremast:

`LOG.error("Unable to find TLS certificate named %s under %s in TLS Cert Template", certificate, env)`